mlflow/R/mlflow/R/model-mleap.R

```diff
@@ -0,0 +1,53 @@
+#' @include model-utils.R
+NULL
+
+#' @rdname mlflow_save_model
+#' @param sample_input Sample Spark DataFrame input that the model can evaluate. This is required by MLeap for data schema inference.
+#'
+#' @export
+mlflow_save_model.ml_pipeline_model <- function(model,
+                                                path,
+                                                model_spec = list(),
+                                                conda_env = NULL,
+                                                sample_input = NULL,
+                                                ...) {
+  if (is.null(sample_input)) {
+    stop("`sample_input` is required by MLeap for data schema inference.")
+  }
+
+  assert_pkg_installed("mleap")
+  if (dir.exists(path)) unlink(path, recursive = TRUE)
+  dir.create(path)
+
+  model_filename <- "model.zip"
+  model_path <- file.path(path, model_filename)
+  mleap::ml_write_bundle(model, sample_input = sample_input, path = model_path)
+  version <- mleap::mleap_installed_versions()$mleap %>%
+    purrr::map(~ numeric_version(.x)) %>%
+    purrr::reduce(~ (if (.x > .y) .x else .y)) %>%
+    as.character()
```

Comment:
I think using `unlist(mleap::mleap_installed_versions()$mleap) %>% max() %>% as.character()` should also work and is probably more readable but it's also taste I guess.

Reply:
I thought so too but unfortunately no: `unlist()` strips the `numeric_version` from the class of the operand, so the rest of the pipe won't work correctly

Reply:
oh right...